src/screens/AddAccounts/02-TokenCurrencyDisclaimer.js

```diff
@@ -64,13 +68,48 @@ export default function AddAccountsTokenCurrencyDisclaimer({
   route,
 }: Props) {
   const { t } = useTranslation();
+  const accounts = useSelector(accountsSelector);
 
   const token = route.params.token;
   const tokenName = `${token.name} (${token.ticker})`;
 
-  function onClose(): void {
+  const parentCurrency = token.parentCurrency;
+
+  const accountData = findTokenAccountByCurrency(token, accounts);
+
+  const parentTokenAccount = accountData ? accountData.parentAccount : null;
+
+  const tokenAccount = accountData ? accountData.account : null;
+
+  const onClose = useCallback(() => {
     navigation.dangerouslyGetParent().pop();
-  }
+  }, [navigation]);
+
+  // specific cta in case of token accounts
+  const onTokenCta = useCallback(() => {
+    if (parentTokenAccount && parentTokenAccount.type === "Account") {
+      onClose();
+      navigation.navigate(NavigatorName.ReceiveFunds, {
+        screen: ScreenName.ReceiveSelectAccount,
+        params: {
+          // prefill search input with account name or token name
+          currency: tokenAccount ? token.name : parentTokenAccount.name,
```

Comment:
Filter the accounts in https://github.com/LedgerHQ/ledger-live-mobile/blob/da9111ce18461c0f60b58ca5a8eff7df609359e6/src/screens/ReceiveFunds/01-SelectAccount.js#L30 if we receive a currency, instead of passing the currency as an initial query for the input search as per https://ledgerhq.atlassian.net/browse/LL-3669

Reply:
I updated it accordingly :+1:  as we discused before with @Arnaud97234  the main issue was to do a similar UX as swap in the filtering and prefilling the sub accounts with the selectedCurrency